source/enc_ctx.c

```diff
@@ -56,8 +56,10 @@ int aws_cryptosdk_enc_ctx_size(size_t *size, const struct aws_hash_table *enc_ct
     size_t serialized_len = 2;  // First two bytes are the number of k-v pairs
     for (struct aws_hash_iter iter = aws_hash_iter_begin(enc_ctx); !aws_hash_iter_done(&iter);
          aws_hash_iter_next(&iter)) {
-        const struct aws_string *key   = iter.element.key;
+        const struct aws_string *key = iter.element.key;
+        AWS_PRECONDITION(aws_string_is_valid(key));
```

Comment:
This isn't really a precondition at this point - maybe just make it a regular assert?

Reply:
Updated!